server/api/utils/builder.py

```diff
@@ -418,7 +419,13 @@ def build_image(
         requirements, commands, with_mlrun, mlrun_version_specifier, client_version
     )
 
-    if not inline_code and not source and not commands and not requirements:
+    if (
+        not inline_code
+        and not source
+        and not commands
+        and not requirements
+        and not force_build
+    ):
         mlrun.utils.logger.info("skipping build, nothing to add")
         return "skipped"
```

Comment:
Id do something more verbose, such as

if force_build:
  logger.debug("forcefully building image")
elif x and not y and not z ..:
  logger.debug("skipping build..")
...